Prevent malformed URLs from blocking the loading of the current tab

MattTheDev opened the pull request:
## Objective

When loading the current tab, the cipher service loads creds matching the URL. It's possible to load malformed URLs into your matching, which then causes the entire tab to stop functioning. 

## Code Changes

**src/misc/utils.ts** The call to getDomain was throwing an error when URL was invalid. Wrapped with try / catch and returned null in the event of issue, following the pattern in the rest of the method.

### Future Enhancement

We should do checking for malformed situations - see https://github.com/bitwarden/browser/issues/1295 for some examples - from being entered.

kspearrin commented:
Are you sure this is related to the issue that you linked? This seems to be try/catching on iterating equivalent domains, not matched URLs.

MattTheDev commented:
Updated the description of the PR, and moved the try / catch. Targetted the correct line to check for issues in parsing - see error:

ERROR Error: Uncaught (in promise): URIError: malformed URI sequence
./node_modules/url/url.js/Url.prototype.parse@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/vendor.js:50717:19
urlParse@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/vendor.js:50606:5
extractHostname@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/vendor.js:49813:19
parse@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/vendor.js:49634:17
getDomain@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/vendor.js:49692:14
getDomain@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/background.js:1097:163
get domain@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/background.js:13802:78
getAllDecryptedForUrl/</<@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/background.js:15814:37
getAllDecryptedForUrl/<@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/background.js:15798:28
fulfilled@moz-extension://bac240a9-576d-3c4f-b3d9-2cd576c5ee65/background.js:15479:58

Actual issue was in utils.ts, when getDomain was getting called. It made it through the other checks - but when trying to get the domain via tldjs library, it would throw the malformed URI error. Rest of the method returns null if invalid - followed that pattern.
